plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/MessageConsoleUtilities.java

```diff
@@ -61,4 +61,31 @@ public static MessageConsole getMessageConsole(String consoleName,
     }
     return messageConsole;
   }
+  
+  public static <C extends MessageConsole> C findOrCreateConsole(String name, ConsoleFactory<C> factory) {
```

Comment:
I wonder if this should be based on names — by using a timestamp for the deploy case, you'll never find a match.  Are we cleaning up old consoles?

An alternative mechanism would be to provide the console class, and / or a predicate to test.  `DeployConsole` could take the `IProject` and use that for comparison purposes.  That would allow having multiple deploys in parallel, but removing old/stale deploy logs.  (Or are older consoles automatically cleaned up?)

(This could be so much nicer with Java 8 with streams.)